proxies-es6/demo.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+// Proxying a normal object
+var target = {};
+var handler = {
+  get(receiver, name = 'world') {
+    return `Hello, ${name}!`;
+  }
+};
+
+var p = new Proxy(target, handler);
+ChromeSamples.log(p.folks);
+
+// Proxying a function object
+function sum(a, b) {
+  return a + b;
+}
+
+var handler2 = {
+  apply: function(target, thisArg, argumentsList) {
+    console.log(`Calculate sum: ${argumentsList}`);
+    return target.apply(thisArg, argumentsList);
+  }
+};
+
+var proxy = new Proxy(sum, handler2);
+ChromeSamples.log(proxy(1, 2));
+
+// Field interception with Proxy and the Reflect API
```

Comment:
What's the Reflect API? Can you link to some relevant docs?

I see that this example differs from the first in that you're setting properties on the `Proxy` instead of a normal `Object`, but I don't understand the motivation for that. Maybe explaining something about the Reflect API would clear that up?